src/entry.rs

```diff
@@ -110,6 +110,23 @@ impl<T: StorableType> Entry<T> {
         }
     }
 
+    #[async_recursion]
+    pub async fn dereference(mut self) -> Result<Entry<T>, CryptoError> {
+        match self.resolved_value.take() {
+            None => match self.value {
+                State::Referenced {
+                    ref path,
+                    ref storer,
+                } => {
+                    let entry = storer.get::<T>(path).await?;
+                    Ok(entry.dereference().await?)
+                }
+                _ => Ok(self)
+            },
+            Some(_) => Ok(self),
```

Comment:
Don't we still need to check if the already filled OnceCell is a reference, and dereference it if so?